kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/coordinatedcommits/CoordinatedCommitsDefaultUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.coordinatedcommits;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.BiFunction;
+
+import io.delta.storage.LogStore;
+import io.delta.storage.commit.actions.AbstractMetadata;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.TableConfig;
+import io.delta.kernel.internal.util.FileNames;
+
+public class CoordinatedCommitsDefaultUtils {
+
+    /**
+     * Write a UUID-based commit file for the specified version to the table at logPath.
+     */
+    public static FileStatus writeCommitFile(
+            LogStore logStore,
+            Configuration hadoopConf,
+            String logPath,
+            long commitVersion,
+            Iterator<String> actions,
+            String uuid) throws IOException {
+        Path commitPath = new Path(
+                FileNames.unbackfilledDeltaFile(
+                        new io.delta.kernel.internal.fs.Path(logPath),
+                        commitVersion,
+                        Optional.of(uuid)).toString());
+        FileSystem fs = commitPath.getFileSystem(hadoopConf);
+        if (!fs.exists(commitPath.getParent())) {
+            fs.mkdirs(commitPath.getParent());
+        }
+        logStore.write(commitPath, actions, false, hadoopConf);
+        return commitPath.getFileSystem(hadoopConf).getFileStatus(commitPath);
+    }
+
+    /**
+     * Get the table path from the provided log path.
+     */
+    public static Path getTablePath(Path logPath) {
+        return logPath.getParent();
+    }
+
+    /**
+     * Helper method to recover the saved value of `tableConfig` from `abstractMetadata`.
+     * Return defaultValue if the key is not in the configuration.
+     */
+    public static <T> T fromAbstractMetadataAndTableConfig(
+            Optional<Engine> engineOpt,
+            AbstractMetadata abstractMetadata,
+            TableConfig<T> tableConfig) {
+        Map<String, String> conf = abstractMetadata.getConfiguration();
```

Comment:
Please convert the abstract metadata to metadata and then invoke fromMetadata on it instead of adding two new APIs in TableConfig.